### PR TITLE
ci(build): replace travis by GitHub Actions

### PR DESCRIPTION
**Pull request by SuperITMan**
ISSUES CLOSED: #29

## PR Checklist

Please check if your PR fulfills the following requirements:

-   [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/code-style/blob/master/CONTRIBUTING.md#-commit-message-guidelines
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [X] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[X] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

All the CI logic is running on travis.

Issue Number: #29 

## What is the new behavior?

CI logic is now running on GitHub Actions.

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by SuperITMan** [comment]
@christophercr @nicanac I gave you access to my repo https://github.com/SuperITMan/code-style like this you can see the result of the Actions :blush:

**Comment by nicanac** [comment]
> @christophercr @nicanac I gave you access to my repo https://github.com/SuperITMan/code-style like this you can see the result of the Actions 😊

Nice, thanks. It's accepted 👍

File: release-publish.sh [hunk 1]
```diff
@@ -80,67 +83,70 @@ logTrace "PROJECT_ROOT_DIR: ${PROJECT_ROOT_DIR}" 1
...
-  fi
 
   if [[ ${NPM_TOKEN} == "" ]]; then
     logInfo "Not publishing because the NPM_TOKEN environment variable is is not defined correctly" 1
-    exit 0;
+    exit 1;
   fi
   
-  # If any of the previous commands in the `script` section of .travis.yaml failed, then abort.
+  # If any of the previous commands in the `script` section of .github/workflows/build.yml failed, then abort.
```
Comment (christophercr) [inline release-publish.sh]: Minor: `script section` -> `steps.run sections`

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  build:
+    name: Test, Build & Release
+    runs-on: "ubuntu-latest"
+    env:
+      GH_ACTIONS_NODE_VERSION: "10"
+      LOGS_DIR: /tmp/code-style/logs
+      LOGS_FILE: /tmp/code-style/logs/build-perf.log
+      TZ: "Europe/Brussels"
+    steps:
+      - uses: actions/checkout@v2
```
Comment (christophercr) [inline .github/workflows/build.yml]: Can you add a comment above this line about where we can find info over `actions/checkout@v2`?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+      - name: Use Node.js 10
+        uses: actions/setup-node@v1
```
Comment (christophercr) [inline .github/workflows/build.yml]: Can you add a comment above this line about where we can find info over `actions/setup-node@v1`?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Test
+        run: npm test
+
+      - name: Build
+        run: npm run build
+
+      - uses: actions/upload-artifact@v1
```
Comment (christophercr) [inline .github/workflows/build.yml]: Can you add a comment above this line about where we can find info over `actions/upload-artifact@v1`?

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        with:
+          name: dist-${{ github.run_id }}
+          path: dist
+
+      - name: Release
+        run: npm run release:publish
+        env:
+          GH_ACTIONS_TAG: ${{ steps.get_tag_name.outputs.TAG_NAME }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (christophercr) [inline .github/workflows/build.yml]: Question... how/where is this NPM defined?

Comment (SuperITMan) [reply .github/workflows/build.yml]: It's simply defined in the repository settings 😊 
immagine

**Comment by SuperITMan** [comment]
@christophercr I applied the changes you mentioned. Could you please have a look ? 😊

**Pull request merged**